File: WorkPlace/Scripts/conf.go
package main

import(
  "fmt"
  "os"
  "os/exec"
  "os/user"
)
func err(e error){
  if e!=nil{panic(e)}
}

func Exec(l string,s string,){
  c := exec.Command("/bin/sh","-c",s+" nvim "+l)
  c.Stdin = os.Stdin
  c.Stdout = os.Stdout
  e:= c.Run()
  err(e)
}

func list(s string){
  user, e:= user.Current()
  err(e)
  switch s {
    case "i3":
      Exec(user.HomeDir+"/.config/i3/config","")
    case "vi":
      Exec(user.HomeDir+"/.config/nvim/init.vim","")
    case "ng":
      Exec("/etc/nginx/nginx.conf","sudo")
    case "ap":
      Exec("/etc/httpd/conf/httpd.conf","sudo")
    case "tm":
      Exec(user.HomeDir+"/.tmux.conf","")
    case "zh":
      Exec(user.HomeDir+"/.zshrc","")
    case "xr":
      Exec(user.HomeDir+"/.Xresources&&xrdb -merge ~/.Xresources","")
    default:
      help()
  }
}

func help(){
  fmt.Print("Specify What you need to confing\n[ng=nginx]\n[ap=apache]\n[i3=i3wm]\n[vi=nvim]\n[ra=ranger]\n[tm=tmux]\n[zh=zsh]\n[xr=xterm]\n")
}

func main(){
  if len(os.Args)>1{
    list(os.Args[1])
  }else{
    help()
  }
}

File: WorkPlace/Scripts/conf.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
char h[] = "Specify What you need to "
           "confing\n[ng=nginx]\n[ap=apache]\n[i3=i3wm]\n[vi=nvim]\n[ra=ranger]\n[tm=tmux]\n[zh=zsh]\n[xr=xterm]\n[co="
           "compton]\n[du=dunst]";
void exec(char x[]) {
  char c[100];
  sprintf(c, "nvim %s", x);
  system(c);
}
void list(char s[]) {
  if (strcmp(s, "i3") == 0) {
    exec("~/.config/i3/config");
  } else if (strcmp(s, "vi") == 0) {
    exec("~/.config/nvim/init.vim");
  } else if (strcmp(s, "ra") == 0) {
    exec("~/.config/ranger/rc.conf");
  } else if (strcmp(s, "ng") == 0) {
    exec("/etc/nginx/nginx.conf&&systemctl restart nginx.service");
  } else if (strcmp(s, "ap") == 0) {
    exec("/etc/httpd/conf/httpd.conf&&systemctl restart httpd.service");
  } else if (strcmp(s, "co") == 0) {
    exec("~/.config/compton/compton.conf");
  } else if (strcmp(s, "du") == 0) {
    exec("~/.config/dunst/dunstrc");
  } else if (strcmp(s, "tm") == 0) {
    exec("~/.tmux.conf");
  } else if (strcmp(s, "zh") == 0) {
    exec("~/.zshrc");
  } else if (strcmp(s, "xr") == 0) {
    exec("~/.Xresources&&xrdb -merge ~/.Xresources");
  } else {
    puts(h);
  }
}
int main(int argc, char* argv[]) {
  if (argc == 2) {
    list(argv[1]);
  } else {
    puts(h);
    return 1;
  }
  return 0;
}
